modules/fluid_properties/include/userobjects/TabulatedFluidProperties.h

```diff
@@ -228,8 +218,34 @@ class TabulatedFluidProperties : public SinglePhaseFluidPropertiesPT
   const SinglePhaseFluidPropertiesPT & _fp;
 
   /// List of required column names to be read
-  const std::vector<std::string> _required_columns{
-      "pressure", "temperature", "density", "enthalpy", "internal_energy"};
+  const std::vector<std::string> _required_columns{"pressure", "temperature"};
+  /// List of possible property column names to be read
+  const std::vector<std::string> _property_columns{
+      "density", "enthalpy", "internal_energy", "viscosity", "k", "cv", "cp", "entropy"};
+  /// Properties to be interpolated entered in the input file
+  MultiMooseEnum _interpolated_properties;
+  /// List of properties read from the data file
+  std::vector<std::string> _read_properties;
+  /// Set of flags to note whether a property is to be interpolated
+  bool _interpolate_density = false;
+  bool _interpolate_enthalpy = false;
+  bool _interpolate_internal_energy = false;
+  bool _interpolate_viscosity = false;
+  bool _interpolate_k = false;
+  bool _interpolate_cp = false;
+  bool _interpolate_cv = false;
+  bool _interpolate_entropy = false;
+
+  /// Index of each property
+  unsigned int _density_idx;
+  unsigned int _enthalpy_idx;
+  unsigned int _internal_energy_idx;
+  unsigned int _viscosity_idx;
+  unsigned int _k_idx;
+  unsigned int _cp_idx;
+  unsigned int _cv_idx;
+  unsigned int _entropy_idx;
+
```

Comment:
These are only set if the property is read from the file / specified in the input, so some of them might never be initialised. Should I give them some default value?

Reply:
That would not hurt anything